### PR TITLE
Optimize mvn dependency download scripts

### PR DESCRIPTION
**Pull request by NvTimLiu**
Move definition of the maven repo into 'version-def.sh', then by overwrite the repo URL
it can download either `SNAPSHOT` cuDF/spark-rapids jars from the internal repo, for the nightly integration tests;
or it can download the `pre-release` jars on the Sonatype repo, for the release verify integration tests. The default maven repo is internal one for the nightly SNAPSHOT integration tests.

### CODE REVIEW

**Comment by NvTimLiu** [comment]
build

**Review by pxLi — DISMISSED** [review]

**Comment by pxLi** [comment]
Looks good if per-merge test and integration tests can pass w/ the script change

**Review by jlowe — DISMISSED** [review]

**Review by revans2 — DISMISSED** [review]

**Comment by NvTimLiu** [comment]
fix merge conflicts

**Comment by NvTimLiu** [comment]
build

File: jenkins/spark-tests.sh
Comment (NvTimLiu) [inline jenkins/spark-tests.sh:35]: Here we diff `PROJECT_VER/PROJECT_REPO` & `PROJECT_TEST_VER/PROJECT_TEST_REPO`.
We only release rapids-4-spark.jar(`PROJECT_VER`) onto Sonatype repo(`PROJECT_REPO`).
rapids-4-spark-udf-examples-`SNAPSHOT`.jar/rapids-4-spark-integration-tests-`SNAPSHOT`.jar(`PROJECT_TEST_VER`) are on internal repo(`PROJECT_TEST_REPO`)

**Comment by pxLi** [comment]
build

**Review by jlowe — APPROVED** [review]

File: jenkins/version-def.sh
Comment (jlowe) [inline jenkins/version-def.sh:32]: Just noticed this typo:
```suggestion
PROJECT_TEST_VER=${PROJECT_TEST_VER:-"0.4.0-SNAPSHOT"}
```

**Review by jlowe — CHANGES_REQUESTED** [review]

**Comment by NvTimLiu** [comment]
build

File: jenkins/version-def.sh
Comment (NvTimLiu) [reply jenkins/version-def.sh]: @jlowe  Thanks a lot!

**Review by jlowe — APPROVED** [review]
Looks good, thanks!

@NvTimLiu in the future, please avoid force-pushing to PRs that are being reviewed.  It makes it difficult to see what has changed since the last time a reviewer looked at the PR and often review comments become disconnected from the code.  It's better to merge up to the latest branch, and as a bonus Github won't dismiss prior approvals if the merge was clean after the approval.

**Comment by NvTimLiu** [comment]
> Looks good, thanks!
> 
> @NvTimLiu in the future, please avoid force-pushing to PRs that are being reviewed. It makes it difficult to see what has changed since the last time a reviewer looked at the PR and often review comments become disconnected from the code. It's better to merge up to the latest branch, and as a bonus Github won't dismiss prior approvals if the merge was clean after the approval.

Got it, thanks!

**Pull request merged**
